backend/de.metas.business/src/main/java/de/metas/project/workorder/undertest/WorkOrderProjectObjectUnderTestService.java

```diff
@@ -0,0 +1,275 @@
+	}
+
+	// public List<WOProjectObjectUnderTest> getWOProjectObjectUnderTestListByOrderLine(@NonNull final OrderLineId orderLineId)
+	// {
+	// 	inOutDAO.retrieveLinesForOrderLine(orderLineBL.getOrderLineById(orderLineId))
+	// 			.stream()
+	// 			.map(iMInOutLine -> )
+	// }
+
+	private void calculatePriceForOrderLine(
+			@NonNull final OrderAndLineId orderAndLineId)
+	{
+		final I_C_Order order = orderBL.getById(orderAndLineId.getOrderId());
+		final I_C_OrderLine orderLine = orderLineBL.getOrderLineById(orderAndLineId.getOrderLineId());
+
+		final SOTrx soTrx = SOTrx.ofBoolean(order.isSOTrx());
+
+		final IEditablePricingContext pricingCtx = pricingBL.createInitialContext(
+				OrgId.ofRepoIdOrAny(order.getAD_Org_ID()),
+				ProductId.ofRepoId(orderLine.getM_Product_ID()),
+				BPartnerId.ofRepoId(order.getC_BPartner_ID()),
+				Quantity.of(orderLine.getQtyEntered(), uomDAO.getById(orderLine.getC_UOM_ID())),
+				soTrx);
+
+		final I_M_PricingSystem pricingSystem = getPricingSystemOrNull(order, soTrx);
+
+		if (pricingSystem == null)
+		{
+			throw new AdempiereException("@NotFound@ @M_PricingSystem_ID@"
+												 + "\n @C_Order_ID@: " + order.getC_Order_ID()
+												 + "\n @C_BPartner_ID@: " + order.getC_BPartner_ID());
+		}
+
+		final PricingSystemId pricingSystemId = PricingSystemId.ofRepoId(pricingSystem.getM_PricingSystem_ID());
+		Check.assumeNotNull(pricingSystemId, "No pricing system found for M_Order_ID={}", order);
+
+		final PriceListId priceListId = priceListDAO.retrievePriceListIdByPricingSyst(
+				pricingSystemId,
+				BPartnerLocationAndCaptureId.ofRepoId(order.getC_BPartner_ID(), order.getC_BPartner_Location_ID()),
+				soTrx);
+		Check.errorIf(priceListId == null,
+					  "No price list found for C_OrderLine_ID {}; C_Order.M_PricingSystem_ID={}, C_Order.C_BPartner_Location_ID={}, C_Order.SOTrx={}",
+					  orderLine.getC_OrderLine_ID(), pricingSystemId, order.getC_BPartner_Location_ID(), soTrx);
+
+		pricingCtx.setPricingSystemId(pricingSystemId);
+		pricingCtx.setPriceListId(priceListId);
+
+		final LocalDate orderedLocalDate = TimeUtil.asLocalDate(orderLine.getDateOrdered(), orgDAO.getTimeZone(OrgId.ofRepoId(order.getAD_Org_ID())));
+		if (orderedLocalDate != null)
+		{
+			pricingCtx.setPriceDate(orderedLocalDate);
+		}
+
+		pricingCtx.setFailIfNotCalculated();
+
+		pricingBL.calculatePrice(pricingCtx);
+	}
+
+	@Nullable
+	private I_M_PricingSystem getPricingSystemOrNull(@NonNull final I_C_Order order, @NonNull final SOTrx soTrx)
+	{
+		if (order.getM_PricingSystem_ID() > 0)
+		{
+			final PricingSystemId pricingSystemId = PricingSystemId.ofRepoId(order.getM_PricingSystem_ID());
+			return priceListDAO.getPricingSystemById(pricingSystemId);
+		}
+
+		final PricingSystemId pricingSystemId = bPartnerDAO.retrievePricingSystemIdOrNull(BPartnerId.ofRepoId(order.getC_BPartner_ID()), soTrx);
+		if (pricingSystemId == null)
+		{
+			return null;
+		}
+
+		return priceListDAO.getPricingSystemById(pricingSystemId);
+	}
```

Comment:
this is not valid, the pricing system should've been a default one coming from a sys config, such that we don't have to add the `0` product price for all customer's pricing systems